### PR TITLE
Edge insertion for CMap/GMap

### PR DESCRIPTION
**Pull request by gdamiand**
_Please use the following template to help us managing pull requests._

## Summary of Changes

Added the function `insert_cell_1_between_two_cells_2()` into `GenericMap` concept, which allows users to insert an edge between two different faces allowing to create faces with holes.

## Release Management

* Affected package(s): CMap/GMap
* Feature/Small Feature (if any): https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/Small_Features/Insert_edge_CMap/GMap
* Link to compiled documentation (obligatory for small feature) [*wrong link name to be changed*](httpssss://wrong_URL_to_be_changed/Manual/Pkg)
* License and copyright ownership:

### CODE REVIEW

**Comment by sloriot** [comment]
Successfully tested in CGAL-6.0-Ic-1

**Comment by gdamiand** [comment]
Now this pr is successfully tested, is someone can have a look to the small feature please? (@sloriot or  @afabri or @someoneelse ?)

File: Combinatorial_map/doc/Combinatorial_map/Concepts/GenericMap.h [hunk 1]
```diff
@@ -917,12 +919,40 @@ If \link GenericMap::are_attributes_automatically_managed `are_attributes_automa
...
 \sa `insert_cell_2_in_cell_3<InputIterator>`
 \sa `remove_cell<i>`
 */
 Dart_descriptor insert_cell_1_in_cell_2(Dart_descriptor d1, Dart_descriptor d2);
 
+/*!
+Inserts a 1-cell between the 2-cell containing `d1` and the one containing `d2`. Returns `previous(d1)`, a descriptor on one dart belonging to the new 1-cell.
+\pre `is_insertable_cell_1_between_two_cells_2(d1,d2)`.
+
+If \link GenericMap::are_attributes_automatically_managed `are_attributes_automatically_managed()`\endlink`==true`, call \link CellAttribute::On_merge `Attribute_type<i>::type::On_merge`\endlink(<I>a</I>,<I>a'</I>) is called for all enabled i-attributes, for i>=2, with <I>a</I> the original 2-attribute associated with `d` and <I>a'</I> the new 2-attribute created during the operation. If set, the dynamic onmerge function of i-attributes is also called on <I>a</I> and <I>a'</I>.
```
Comment (afabri) [inline Combinatorial_map/doc/Combinatorial_map/Concepts/GenericMap.h:933]: Note that a prime looks like  you had forgotten one tick.

File: Combinatorial_map/doc/Combinatorial_map/Concepts/GenericMap.h [hunk 2]
```diff
@@ -917,12 +919,40 @@ If \link GenericMap::are_attributes_automatically_managed `are_attributes_automa
...
+\sa `insert_cell_0_in_cell_1`
+\sa `insert_cell_0_in_cell_2`
+\sa `insert_cell_1_in_cell_2`
+\sa `insert_dangling_cell_1_in_cell_2`
+\sa `insert_cell_2_in_cell_3<InputIterator>`
+\sa `remove_cell<i>`
+*/
+Dart_descriptor insert_cell_1_between_two_cells_2(Dart_descriptor d1, Dart_descriptor d2);
+
+/*! Call `insert_cell_1_in_cell_2` if `is_insertable_cell_1_in_cell_2(d1, d2)`, otherwise call `insert_cell_1_between_two_cells_2`.
```
Comment (afabri) [inline Combinatorial_map/doc/Combinatorial_map/Concepts/GenericMap.h:949]: I would add a parenthesis pair after the function name.   And if it was only for getting a link.   I would do the same for all the the '\sa`.  If I remember correctly @lrineau finds it correct not to put parenthesis.

File: Combinatorial_map/doc/Combinatorial_map/Concepts/GenericMap.h [hunk 3]
```diff
@@ -976,6 +1008,17 @@ This is possible if `d1`\f$ \neq \f$ `d2` and `d1` can be reached from `d2` by u
 */
 bool is_insertable_cell_1_in_cell_2(Dart_const_descriptor d1, Dart_const_descriptor d2);
 
+/*!
+Returns true iff it is possible to insert a 1-cell in the generic map between `d1` and `d2`.
```
Comment (afabri) [inline Combinatorial_map/doc/Combinatorial_map/Concepts/GenericMap.h:1012]: true should be backticked.

File: Combinatorial_map/doc/Combinatorial_map/Concepts/GenericMap.h [hunk 4]
```diff
@@ -976,6 +1008,17 @@ This is possible if `d1`\f$ \neq \f$ `d2` and `d1` can be reached from `d2` by u
...
+
+This is possible if `d1`\f$ \neq \f$ `d2` and `d1` can not be reached from `d2` by using some `previous` and `next` calls.
```
Comment (afabri) [inline Combinatorial_map/doc/Combinatorial_map/Concepts/GenericMap.h:1014]: We had a discussion about mathmode vs code for not equal.   I prefer code.

**Comment by gdamiand** [comment]
> [Jun 20, 2023](#event-9577691155)
> [@github-actions](/apps/github-actions)
> ###

@github-actions modifications are only about documentation, tested label is still valid ;)

File: Combinatorial_map/doc/Combinatorial_map/Concepts/GenericMap.h (hunk 4 above)
Comment (gdamiand) [reply Combinatorial_map/doc/Combinatorial_map/Concepts/GenericMap.h:1014]: you want <code>!=</code> ?

File: Combinatorial_map/doc/Combinatorial_map/Concepts/GenericMap.h (hunk 1 above)
Comment (gdamiand) [reply Combinatorial_map/doc/Combinatorial_map/Concepts/GenericMap.h]: Sorry I don't see where?

**Comment by gdamiand** [comment]
Thanks @afabri for your review !!

**Comment by lrineau** [comment]
> > [Jun 20, 2023](#event-9577691155)
> > [@github-actions](/apps/github-actions)
> 
> @github-actions modifications are only about documentation, tested label is still valid ;)

Our testsuite also test the compilation of the documentation. Your changes to the document may trigger new errors.

@sloriot Can you please test this PR again?

File: Combinatorial_map/doc/Combinatorial_map/Concepts/GenericMap.h (hunk 1 above)
Comment (afabri) [reply Combinatorial_map/doc/Combinatorial_map/Concepts/GenericMap.h]: You have `a` and `a'`

**Comment by afabri** [comment]
> Now this pr is successfully tested, is someone can have a look to the small feature please? (@sloriot or  @afabri or @someoneelse ?)

Note that there are more remarks in the Discussion section of the small feature.

File: Combinatorial_map/doc/Combinatorial_map/Concepts/GenericMap.h (hunk 1 above)
Comment (gdamiand) [reply Combinatorial_map/doc/Combinatorial_map/Concepts/GenericMap.h]: you prefer another name? for example a and a2 ?

File: Combinatorial_map/doc/Combinatorial_map/Concepts/GenericMap.h (hunk 4 above)
Comment (gdamiand) [reply Combinatorial_map/doc/Combinatorial_map/Concepts/GenericMap.h:1014]: Indeed I use \f$ \neq \f$ in many other places; thus it is probably better to keep it.

File: Combinatorial_map/doc/Combinatorial_map/Concepts/GenericMap.h (hunk 1 above)
Comment (afabri) [reply Combinatorial_map/doc/Combinatorial_map/Concepts/GenericMap.h]: Let's keep it as the prime really gives the idea of that it is the same thing before/after.

**Review by afabri — APPROVED** [review]

**Comment by sloriot** [comment]
Successfully tested in CGAL-6.0-Ic-15

**Pull request merged**